Further serial gimbal/headtracker improvements

mmosca opened the pull request:
A follow up to #10109, for further improvement.

* [x] Add trim to gimbal to allow Uptilt on quad setups. This only applies to locked center mode.
```
set gimbal_pan_trim = 0
set gimbal_tilt_trim = 0
set gimbal_roll_trim = 0
```

* [x] Integrate with OSD pan compensation

mmosca committed:
Merge remote-tracking branch 'origin/master' into mmosca-gimbal-test

mmosca committed:
Add axis trim

mmosca committed:
Merge remote-tracking branch 'origin/master' into mmosca-gimbal-test

mmosca committed:
Add trim when headtracker info is stale.

Stale headtracker info results in center position of gimbal.

mmosca committed:
fix test build

mmosca committed:
Merge remote-tracking branch 'origin/master' into mmosca-gimbal-test

mmosca committed:
Take gimbal input into account for pan servo compensation and center
indication

mmosca committed:
Cleanup

mmosca committed:
Use printf for LOG_DEBUG in sitl.

MrD-RC commented on src/main/io/osd.c:
```diff
@@ -1203,8 +1204,15 @@ int16_t osdGetHeading(void)
+    gimbalDevice_t *dev = gimbalCommonDevice();
+    if (dev && gimbalCommonIsReady(dev)) {
+        servoPosition = gimbalCommonGetPanPwm(dev);
+        servoMiddle = PWM_RANGE_MIDDLE;
```
One question. Shouldn't this use gimbal_pan_trim for the central position?

mmosca replied:
Makes sense. I updated everything to use trim, (except, raw headtracker).

If no valid headtracker is found, behavior is the same as "center lock" as well.

If there is pan trim, osd will reflect what happens for non head tracked modes as well, so you may need to look a bit sideways to center it in osd. (looking the same direction as non head tracked mode).

mmosca committed:
Merge remote-tracking branch 'origin/master' into mmosca-gimbal-test

mmosca committed:
Add pan trim to stick inputs and center checks

Add pan trim to rc inputs so it can match center flight mode.

Also add pan trim to osd center checks.

I did not add pan trim to headtracker output, as that would probably
confuse the user more than not.

And not adding it to the headtracker input, the osd will still match the expected position.

You probably shouldn't be mounting your gimbal sideways, anyway. ;)

mmosca committed:
Fix center when no valid headtracker data is available
